### DISCUSSION TITLE
Non exhaustive variant combinations in recipe

### DISCUSSION THREAD

**Discussion by shulcsm** [vanilla-extract-css/vanilla-extract#1174] [Q&A] [answered]
I have some compound type variations that are not compatible.
```ts
type HorizontalMenuVariants = {
  variant: 'tabs' | 'pills';
  tone: 'primary' | 'secondary';
  justified: boolean;
};

type VerticalMenuVariants = {
  variant: 'static' | 'collapse' | 'pop-out';
  tone: 'light' | 'dark';
};

type MenuVariants = HorizontalMenuVariants | VerticalMenuVariants;
```

Is there a way to describe it with recipe?   
Best thing i can come up is two recipes with type guard on variant and runtime toggle

**Comment by askoufis** [comment, marked as answer]
I don't think there's a way to do it with a single recipe. IMO a single recipe and a single component go well together, so the fact that you've had to create two separate recipes could be a signal that really you should be creating two components here, a `VerticalMenu` and a `HorizontalMenu`, rather than toggling between recipes at runtime based on props.
